forc-pkg/src/pkg.rs

```diff
@@ -99,6 +100,9 @@ pub struct BuiltPackage {
     source_map: SourceMap,
     pub pkg_name: String,
     pub built_pkg_descriptor: BuiltPackageDescriptor,
+    /// If this pkg is a contract dependency of another pkg in the build graph, this points to the
+    /// version of this pkg compiled without tests.
+    pub pkg_without_tests: Option<Arc<BuiltPackage>>,
```

Comment:
I am not really happy how this turned out, I wanted to break `BuiltPackage` into `BuiltContract` and `BuiltNonContract` to overcome this but I remember having a discussion not do that to keep different logic of building for different program types in only `forc-pkg`.

Reply:
Rather than storing this inside of `BuiltPackage` and creating a recursive data structure, perhaps we can just store a second `BuiltPackage` in the `built_packages` `Vec` that is produced by the `build` function?

Reply:
Sounds good, I ended up retuning `Vec` of `(BuiltPackage, Option<BuiltPackage>)` as non-contract members are not recompiled without tests.